plugins/inputs/mesos/mesos.go

```diff
@@ -315,6 +321,12 @@ func getMetrics(role Role, group string) []string {
 			"master/outstanding_offers",
 		}
 
+		// These groups are empty because filtering is done in gatherMainMetrics
+		// based on presence of "framework_offers"/"allocator" in MasterCols.
+		// These lines are included to prevent the "unknown" info log below.
+		m["framework_offers"] = []string{}
+		m["allocator"] = []string{}
```

Comment:
Is there a reason why we don't list the metrics here and remove the logic in gatherMainMetrics?

Reply:
Allocator and framework metric names may include values that can't be predicted, such as framework ID and role (docs [here](https://mesos.apache.org/documentation/latest/monitoring/#frameworks) and [here](https://mesos.apache.org/documentation/latest/monitoring/#allocator)). That prevents us from listing all the possible allocator and framework metrics here.